src/Storages/MergeTree/MergeTreeDeduplicationLog.h

```diff
@@ -0,0 +1,180 @@
+#pragma once
+#include <Core/Types.h>
+#include <common/StringRef.h>
+#include <IO/WriteBufferFromFile.h>
+#include <Storages/MergeTree/MergeTreePartInfo.h>
+#include <map>
+#include <list>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+namespace DB
+{
+
+/// Description of dedupliction log
+struct MergeTreeDeduplicationLogNameDescription
+{
+    /// Path to log
+    std::string path;
+
+    /// How many entries we have in log
+    size_t entries_count;
+};
+
+/// Simple string-key HashTable with fixed size based on STL containers.
+/// Preserves order using linked list and remove elements
+/// on overflow in FIFO order.
+template <typename V>
+class LimitedOrderedHashMap
```

Comment:
LRUCache?

Reply:
But it's not the last recently used map, it's the last recently added map.